src/myComponents/table/SearchResultBody.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react";
+
+import Moment from "moment";
+
+export default function SearchResultBody({ showProfile, item, theme }) {
+  const [selected, setSelected] = useState(false);
+  let checckDiff = (InDate, OutDate) => {
+    const checkInDate = Moment(InDate, "YYYY-MM-DD");
+    const checkOutDate = Moment(OutDate, "YYYY-MM-DD");
+    const diff = checkOutDate.diff(checkInDate, "days");
+    return diff;
+  };
+
+  return (
+    <tr
+      style={{
+        backgroundColor: `${selected ? "#f8d7da" : "rgb(238, 238, 238)"}`,
+        cursor: "pointer"
+        // ...theme,
```

Comment:
I uncommented this line (and fixed the comma) in order to use the ThemeContext as you aimed):
 ```suggestion
        cursor: "pointer",
        ...theme
```

However, as you mentioned, when I did that, the feature which highlights a row after each click doesn't work anymore.
The reason for this is you are adding the `...theme` on the last line which overrides your other style on line 17. Basically your line 17 is redundant as it gets overwritten all the time.

In general, your ThemeContext (or any context) should not enforce how a component should behave. It should just simply give a context (or more information about the way a component should be rendered). I would suggest, instead of determining the background colour from the top to down, the ThemeContext can simply inform if a component should be displayed as dark or light. Then the component can decide which colour to use.